src/vulkan/graphics_pipeline.cc

```diff
@@ -327,32 +328,38 @@ Result GraphicsPipeline::Initialize(uint32_t width,
   return {};
 }
 
-Result GraphicsPipeline::SetBuffer(BufferType type,
-                                   uint8_t location,
-                                   const Format& format,
-                                   const std::vector<Value>& values) {
+Result GraphicsPipeline::SetVertexBuffer(BufferType type,
+                                         uint8_t location,
+                                         const Format& format,
+                                         const std::vector<Value>& values,
+                                         VertexBuffer* vertex_buffer) {
   // TODO(jaebaek): Handle indices data.
-  if (type != BufferType::kVertex)
-    return {};
+  if (type != BufferType::kVertex) {
+    return Result(
+        "GraphicsPipeline::SetVertexBuffer: non-vertex buffer type not "
+        "implemented");
+  }
 
-  if (!vertex_buffer_)
-    vertex_buffer_ = MakeUnique<VertexBuffer>(device_);
+  if (vertex_buffer == nullptr) {
+    return Result(
+        "GraphicsPipeline::SetVertexBuffer: vertex buffer is nullptr");
+  }
 
   DeactivateRenderPassIfNeeded();
 
   Result r = command_->BeginIfNotInRecording();
   if (!r.IsSuccess())
     return r;
 
-  vertex_buffer_->SetData(location, format, values);
+  vertex_buffer->SetData(location, format, values);
   return {};
 }
 
-Result GraphicsPipeline::SendBufferDataIfNeeded() {
-  if (!vertex_buffer_)
+Result GraphicsPipeline::SendBufferDataIfNeeded(VertexBuffer* vertex_buffer) {
```

Comment:
Why does this take a VertexBuffer? Can't we send other types of buffer data as well?

Reply:
Changed it as `SendVertexBufferDataIfNeeded`